compatibility changes for Neos v.3.3

Magarif opened the pull request.

remuslazar commented on Classes/Command/NodeCommandController.php:
```diff
@@ -14,7 +14,7 @@
 use Doctrine\ORM\Query;
 use Neos\Flow\Annotations as Flow;
-use Neos\Flow\Object\ObjectManagerInterface;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
```
This is a quote serious error and not a "minor phpstorm error"? I mean, just using the old, now obsolete and non existent path will lead to a fatal error.

remuslazar commented on Classes/Command/NodeCommandController.php:
```diff
@@ -516,6 +517,7 @@ private function searchTermMatch($term, $value)
      * @param NodeInterface[]|array[] $nodes
      * @param string $indend
      * @param string $propertyName which property to display, defaults to title
+     * @throws \Neos\ContentRepository\Exception\NodeException
```
I will not mix this kind-of stuff in this PR. This PR should only address the upgrade to new Flow/Neos versions and not also do the cleanups to make the new phpStorm defaults happy. This has nothing to do with the upgrade itself.

remuslazar replied:
in the meantime, just disable the warning in phpstorm, e.g. using:

<img width="914" alt="image" src="https://user-images.githubusercontent.com/6108134/39622945-40eb6898-4f94-11e8-8bef-7cffab9ae914.png">

Commit:
TASK: Apply migration TYPO3.FLOW3-201209201112

Move all code to PSR-0 compatible directory structure, remove
Package.xml,

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.FLOW3-201206271128

Replace DataNotSerializeableException with DataNotSerializableException.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.FLOW3-201205292145

Replace FileTypes use with MediaTypes use.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.FLOW3-201201261636

Adjust to the major changes in FLOW3 1.1 when coming from 1.0.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-201209251426

Rename FLOW3 to TYPO3 Flow

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Fluid-201205031303

Rename form.textbox to form.textfield

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-20141218134700

Rename node property type 'date' to DateTime

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Fluid-20141113120800

Rename the "resource" argument of the security.ifAccess ViewHelper to
"privilegeTarget"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TYPO3CR-140911160326

Adjust to renamed NodeLabelGeneratorInterface.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TYPO3CR-130523180140

Adjust to removed TYPO3.ContentRepository:Folder node type by replacing
it

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-201212051340

Change entity resource definitions from using _ to \

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Fluid-20141121091700

Warn about removed ReflectionService dependency from AbstractViewHelper

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-201410010000

Migrates the former HtmlEditor to CodeEditor

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-201409071922

Adjust to updated folder name for TypoScript in site packages

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TypoScript-130516235550

Rename "renderTypoScript" VH to just "render"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-201407061038

Adjust to updated date format for inspector date editor

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-20141113121400

Adjust "Policy.yaml" to new syntax

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-20141113115300

Rename setting TYPO3.Neos.modules.<moduleName>.resource to
"privilegeTarget"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos.NodeTypes-201309111655

Change node type and TS object names in NodeTypes.yaml and PHP code.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TypoScript-130516234520

Rename "renderTypoScript" VH to just "render"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-201405111147

This migration does not actually change any code. It just displays a
warning if a TypeConverters canConvertFrom() is

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-201310031523

Change Neos\Flow\Persistence\Doctrine\DatabaseConnectionException to

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-201211151101

Check for globally defined role identifiers in Policy.yaml files

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TYPO3CR-141101082142

Adjust namespaces for ContentRepository related FlowQuery operations
that were moved from TYPO3.Neos to TYPO3.TYPO3CR

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-20161115140430

Adjust to the renaming of the Object namespace in Flow 4.0

Commit:
TASK: Apply migration Neos.Media-20161124233100

Adjusts code to package renaming from "TYPO3.Media" to "Neos.Media"

Commit:
TASK: Apply migration Neos.Eel-20161124230101

Adjusts code to Eel Renaming

Commit:
TASK: Apply migration Neos.Flow-20161124204700

Adjusts code to package renaming from "TYPO3.Flow" to "Neos.Flow"

Commit:
TASK: Apply migration TYPO3.Flow-20151113161300

Adjust "Settings.yaml" to new "requestPattern" and "firewall" syntax
(see FLOW-412)

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Flow-20161124224015

Adjusts code to cache extraction

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Twitter.Bootstrap-20161124204912

Adjusts code to package renaming from "TYPO3.Twitter.Bootstrap" to
"Neos.Twitter.Bootstrap"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Setup-20161124230842

Adjusts code to Setup Renaming

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Flow-20161124204701

Adjusts code to package renaming from "Neos.Flow.Utility.Files" to
"Neos.Utility.Files" and other extractions of the "Utility" packages.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Neos-20150303231600

Migrates ImageVariant to ImageInterface

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Form-20160601101500

Adjust "Settings.yaml" to use validationErrorTranslationPackage instead
of translationPackage

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Form-20161124205254

Adjusts code to package renaming from "TYPO3.Form" to "Neos.Form"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Fluid-20150214130800

Warn about "escapeOutput" property for existing ViewHelpers to ensure
backwards-compatibility

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Imagine-20161124231742

Adjusts code to Imagine Renaming

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.TYPO3CR-20150510103823

Migrates to new super type syntax in node type configuration

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Party-20161124225257

Adjusts code to package renaming from "TYPO3.Party" to "Neos.Party"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.Flow-20161115140400

Adjust to the renaming of the Resource namespace and class in Flow 4.0

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.ContentRepository-20161125012000

Adjusts code to package renaming from "TYPO3.TYPO3CR" to
"Neos.ContentRepository"

Commit:
TASK: Apply migration Neos.Neos-20161125002322

Adjusts code to package renaming from "TYPO3.Neos" to "Neos.Neos"

Commit:
TASK: Apply migration Neos.Neos-20161201222211

Migrate namespaces for fusion core implementation and helper classes

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20161125122412

Allow to migrate Sites.xml files

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Fusion-20161125104701

Adjusts code to package renaming from "TYPO3.TypoScript" to
"Neos.Fusion" in Fusion files

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Fusion-20161125013710

Adjusts code to package renaming from "TYPO3.TypoScript" to
"Neos.Fusion"

Commit:
TASK: Apply migration Neos.Fusion-20161219092345

Migrate name for the Fusion cache to Neos_Neos_Fusion

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20161125104802

Adjusts code to package renaming from "TYPO3.Neos" to "Neos.Neos" in
Fusion files

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Setup-20161125014759

Adjusts settings path from TYPO3.Setup to Neos.Setup to Setup Renaming

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Flow-20161125124112

Adjusts code to Neos\Flow\Utility\Unicode adjustment

Commit:
TASK: Apply migration Neos.Fusion-20161202215034

Migrate name for the Fusion content cache to Neos_Fusion_Content

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.NodeTypes-20161125104800

Adjusts code to package renaming from "TYPO3.Neos.NodeTypes" to
"Neos.NodeTypes" in Fusion files

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.NodeTypes-20161125002300

Adjusts code to package renaming from "TYPO3.Neos.NodeTypes" to
"Neos.NodeTypes"

Commit:
TASK: Apply migration Neos.ContentRepository.Search-20161210231100

Adjusts code to package renaming from "TYPO3.TYPO3CR.Search" to
"Neos.ContentRepository.Search"

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Fusion-20161201202543

Moves fusion files from old path ``Resources/Private/TypoScript/`` to
the new path

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.SwiftMailer-20161130105617

Adjusts code to package renaming from "TYPO3.SwiftMailer" to
"Neos.SwiftMailer".

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration TYPO3.FluidAdaptor-20161130112935

Adjusts code to package renaming from "TYPO3.Fluid" to
"Neos.FluidAdaptor".

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Fusion-20170120013047

Migrate name for the TypoScriptView to FusionView

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Seo-20170127154600

Migrate to new namespace

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Flow-20170127183102

Migrate bootstep names.

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20170115114620

Migrate usages of the Settings path
Neos.Flow.security.authentication.providers.Typo3BackendProvider to
Neos.Flow.security.authentication.providers[Neos.Neos:Backend]

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.ContentRepository-20161219093512

Migrate name for the CR node type configuration

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Flow-20170125103800

Migrate usages of the path [TYPO3][Flow][Security][Authentication] to
[Neos][Flow][Security][Authentication]

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20161220163741

Migrate usages of the Settings path Neos.Neos.typoScript to
Neos.Neos.fusion

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Media-20161219094126

Migrate name for the media image size cache

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20161219122512

Migrate usages of TypoScriptService to FusionService

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Neos-20161219094403

Migrate several cache keys for the Neos.Neos package

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
TASK: Apply migration Neos.Fusion-20161219130100

Migrate name for the TypoScriptView to FusionView

Note: This migration did not produce any changes, so the commit simply
marks the migration as applied. This makes sure it will not be applied
again.

Commit:
composer changes

remuslazar commented:
@Magarif looks good to me. There are still some references to TYPO3 in the README and inline comments but we can leave them as is for now, this will not hurt much.

remuslazar reviewed (approved):
👍
